### ISSUE TITLE
ontology with prefixes analyzed and displayed incorrectly

### ISSUE THREAD

**Issue by mcodescu** [spechub/Hets#1880]
With the ontology
```
logic OWL

ontology O1 = 
 Prefix:   : <http://test.org/O1#>
 Prefix:  a: <http://test.org/A#>
 Prefix:  b: <http://test.org/B#>
 
 Class: C
 Class: a:C
 Class: b:C
end
```
the result of `Show theory` is
```
logic OWL.ELQLRL-ALC

Prefix: : <<http://test.org/O1#>>
Prefix: a: <<http://test.org/A#>>
Prefix: b: <<http://test.org/B#>>

Class: C
```
Two issues: 
 - double < and > for prefixes
 - only one class C instead of C with different prefixes.

**Comment by mcodescu** [comment]
For the issue with the double angles: the IRI of the value of the prefix has the flag `hasAngles` set to `true` (note that if the angles are missing we will correctly get a parsing error). Thus, when displayed, the method `iriToStringFull` adds the first pair of angles. The method `printPrefixes` in `OWL2/ManchesterPrint.hs` always prints angles, so thus we get the second pair. 

Unless we have an example when adding the second pair of angles is needed, I would not print them the second time.

**Comment by mcodescu** [comment]
For the second issue: the static analysis tries to add `a:C` to the signature where `C` already exists. Checking equality is done via the case for full IRIs, when the `prefixName`s are not compared. Adding this comparison gets things to work. But I wonder if not rather `hasFullIRI` should be corrected, also because when showing the resulting signature the IRIs are again identified as full and for all three of them we get `C` instead of `C, a:C, b:C`, as the `prefixName` is not displayed for full IRIs.

**Comment by tillmo** [comment]
Your solution for the first issue seems plausible to me.
Concerning the second issue, indeed the three different names should be displayed differently. I do not understand why the `prefixName` is not displayed for full IRIs. I think either the full IRI has to be displayed, or (if it is decomposed into prefix and local path) then the CURIE (= prefix + local path) generating the full IRI.

**Comment by mcodescu** [comment]
My question was: is `a:C` a full IRI (with `isAbbrev` set to `True`) or an IRI that is not full? 

If it is a full IRI, we must make two changes: comparison between two IRIs should take into account the prefix and displaying of full IRIs should take into account the case when `isAbbrev` holds. 

If it is not a full IRI (my intuition), then `hasFullIri` should no longer identify `a:C` as a full IRI.

**Comment by mcodescu** [comment]
I get the expected results with `&& not (isAbbrev i)` added as a condition to using `showIRIFull` in `showIRI` and with `prefixName` added in comparison of IRIs. An alternative to the first would be modifying `iriToStringFull`.

**Comment by mcodescu** [comment]
Now, the reason why `printPrefixes` added the angles in the first place: 
```
%prefix( a: <http://test.org/test/A#>)%

ontology O2 = 
 Class: a:C
end
```
yields
```
logic OWL.ELQLRL-ALC

Prefix: a: http://test.org/test/A#

Class: a:C
```
and the prefixes are now missing the required angles. 

The DOL and OWL prefixes should be consistent. One mismatch is that DOL prefixes are saved as a map from String to IRI and OWL prefixes as a map from String to String. We can surely live with this, but we must make sure that the same convention is followed: either we have the angles as part of the value or we don't.

**Comment by mcodescu** [comment]
I just pushed a fix for this on the branch `iris`. A simple test is 
```
%prefix( a: <http://test.org/test/A#>)%

logic OWL

ontology O2 = 
 Prefix:  b: <http://test.org/B#>
 Class: C
 Class: a:C
 Class: b:C
end
``` 
I'd wait for #1882, #1891 and perhaps a refactoring when OWL prefixes are IRIs instead of strings before merging.

**Comment by mcodescu** [comment]
No, prefixes are IRIs and they expand to strings.

**Comment by mcodescu** [comment]
Solved by #1909
